File: PathStructure.WatcherHost/Dtos/WatcherAbortedNotificationParams.cs
namespace PathStructure.WatcherHost.Dtos
{
    /// <summary>
    /// Represents watcher aborted notification parameters.
    /// </summary>
    internal sealed class WatcherAbortedNotificationParams
    {
        /// <summary>
        /// Gets or sets the aborted message.
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// Gets or sets the error details.
        /// </summary>
        public string Error { get; set; }

        /// <summary>
        /// Gets or sets the ISO timestamp for the aborted event.
        /// </summary>
        public string Timestamp { get; set; }
    }
}

File: PathStructureClient/src/dtos/WatcherAbortedNotificationParams.js
class WatcherAbortedNotificationParams {
  /**
   * @param {object} options
   * @param {string} options.message
   * @param {string} [options.error]
   * @param {string} options.timestamp
   */
  constructor({ message, error, timestamp }) {
    /** @type {string} */
    this.message = message;
    /** @type {string|undefined} */
    this.error = error;
    /** @type {string} */
    this.timestamp = timestamp;
  }
}

module.exports = { WatcherAbortedNotificationParams };
